crates/wysiwyg/src/composer_model/replace_text.rs

```diff
@@ -91,7 +93,32 @@ where
             self.state.end = self.state.start;
             self.create_update_replace_all()
         } else {
-            panic!("Unexpected multiple nodes on a 0 length selection")
+            // Special case, there might be one or several empty text nodes at the cursor position
+            let empty_text_leaves: Vec<&DomLocation> = leaves
+                .into_iter()
+                .filter(|l| {
+                    if let DomNode::Text(t) =
+                        self.state.dom.lookup_node(&l.node_handle)
+                    {
+                        t.data().is_empty()
+                    } else {
+                        false
+                    }
+                })
+                .collect();
+            for (i, leaf) in empty_text_leaves.iter().enumerate().rev() {
+                if i == 0 {
+                    self.state.dom.replace(
+                        &leaf.node_handle,
+                        vec![DomNode::new_line_break()],
+                    );
+                } else {
+                    self.state.dom.remove(&leaf.node_handle);
+                }
+            }
+            self.state.start += 1;
+            self.state.end = self.state.start;
+            self.create_update_replace_all()
```

Comment:
This function is getting long, so I'd prefer this to be a separate function.

But if you're rushing to get this out, I'm happy for you to make that change later.